### PR TITLE
Add HeadlessInstallation feature gate to disable API/UI/Ingress

### PR DESCRIPTION
**Pull request by xrstf**
**What does this PR do / Why do we need it**:
Now that most of the bare minimum resources work just fine with GitOps and we have a dedicated gitops e2e test, it's now time to strip down the gitops environment even more and allow to disable nginx/dex/cert-manager entirely.

To achieve this, I chose to add a feature gate to KKP. Determining the headless state just based on "are ui.replicas=0 and api.replicas=0?" seems brittle and could lead to unintended consequences. With a dedicated feature gate, the admins can make their intentions absolutely clear and the operator can take the appropriate actions.

Note that this leaves behind a KKP system that has **no user contexts**. Meaning that only cluster-admins with access to master/seed clusters can make use of KKP. User authentication is still a giant TBD epic.

**Does this PR introduce a user-facing change?**:
```release-note
A new `HeadlessInstallation` (preview, not yet ready for production) feature flag can be used to disable the KKP UI, API and Ingress, which will also skip installing nginx/Dex/cert-manager. Use this if you intend to only access the master/seed clusters directly and need no user separation.
```

### CODE REVIEW

**Comment by embik** [comment]
This looks good to me, do you want me to lgtm it?

**Review by ahmedwaleedmalik — APPROVED** [review]
Good stuff 😁 

/approve 🚀 
/hold if other reviews are required

**Comment by xrstf** [comment]
/hold cancel

#yolo

**Pull request merged**
